### ISSUE TITLE
output NaN value when training

### ISSUE THREAD

**Issue by liu2048** [zhenyuanlu/PainAttnNet#1]
Hi, your paper and code are very helpful.

I try to reproduce your results, but in training step, just in Fold 1 and epoch 1, the model outputs all NaN values (shape is [72, 2]).

Why is this happen, how can i fix it?

I will be really appreciate for your response!

**Comment by zhenyuanlu** [comment]
Hi Liu - Thanks for reaching out! Regarding the NaN outputs you're seeing, it’s possible that this is due to some instability at the very beginning of training, which might stabilize as the model continues to train. Since softmax is involved, it could initially produce extreme values that cause this behavior, but as training progresses, it should correct itself.

I’d recommend letting the model run for a few more epochs to see if the NaN issue resolves itself. If the problem persists beyond the initial stages, it might help to adjust a few settings:

- Lower the learning rate
- Check if the input data contains any NaN or Inf values

**Comment by liu2048** [comment]
Thanks for your reply @zhenyuanlu !

I have tried a lower learning rate (0.001 -> 0.00001) and more training epochs (1 -> 20), and the output result of all rounds are all NaN values. 

config.json is

``` 
{
    "name": "LOSO_0_vs_4",
    "n_gpu": 1,
    "arch": {
        "type": "PainAttnNet",
        "args": {
        }
    },
    "data_loader": {
        "args":{
            "batch_size": 128,
            "num_folds": 86
        }
    },
    "optimizer": {
        "type": "Adam",
        "args":{
            "lr": 0.00001,
            "weight_decay": 0.001,
            "amsgrad": true
        }
    },
    "metrics": [
        "accuracy"
    ],
    "trainer": {
        "epochs": 20,
        "save_dir": "saved/",
        "save_period": 30,
        "verbosity": 2,
        "monitor": "min val_loss"
    },
    "label_converter": {
        "0": 0,
        "1": -1,
        "2": -1,
        "3": -1,
        "4": 1
    }
}

```
![image](https://github.com/user-attachments/assets/a3798c7a-a273-49a4-88fe-58ab7bff8da1)


I tried other BioVid biosignals, and interestingly, only GSR (EDA) had this problem, ECG and EMG did not have NaN outputs. I checked the differences between GSR and other two signals, GSR seems to have smaller variance and flatter waveforms.

**Comment by liu2048** [comment]
I tried more experiments on ECG. Validation loss and validation accuracy seem to be less than ideal, especially since validation loss is known to be maintained at 0.5, that is, 0 or 1 for all validation samples in binary classification. Is this some kind of overfitting, or is there something wrong with the model itself ?
Fold 1
![image](https://github.com/user-attachments/assets/3354b98c-ae2e-46d5-9f57-ff524a29d396)
Fold 5
![image](https://github.com/user-attachments/assets/4f2fb42b-d089-4f6f-8494-c69fa4afe332)
Fold 10
![image](https://github.com/user-attachments/assets/041251cc-059d-433e-b904-c0da27050ea8)

**Comment by zhenyuanlu** [comment]
This seems like overfitting to me, but the accuracy plot suggests that there’s no improvement at all.

- Please follow the paper guideline, set leave-one-out cross-validation (LOOCV), which means set $k$=87. $k\in${1,5,10} won't work well for this BioVid dataset as the data nature has 87 individual subjects.

> We used 87-fold cross-validation for the BioVid by splitting the subjects into 87 groups, therefore, each subject is in one group as a leave-one-out cross-validation (LOOCV). We trained the model on 86 subjects and tested it on one subject with 100 epochs for each iteration. Ultimately, the macro performance matrices were computed by combining the projected pain intensity classes from all 87 iterations. [Lu, 2023 - 4.2 Experimental settings](https://www.frontiersin.org/journals/physiology/articles/10.3389/fphys.2023.1294577/full)

- Make sure you’ve installed the correct versions of the dependencies as listed in the README.
- You might want to try adjusting the learning rate to 1e-03.
- Set epoch as 100
